sdks/python/apache_beam/testing/test_stream_test.py

```diff
@@ -410,17 +395,30 @@ def process(self,
         yield (elm, ts, side)
 
     records = (main_stream     # pylint: disable=unused-variable
-               | beam.ParDo(RecordFn(), beam.pvalue.AsList(side_stream))
-               | beam.Map(recorded_elements))
+               | beam.ParDo(RecordFn(), beam.pvalue.AsList(side_stream)))
     p.run()
 
-    # TODO(BEAM-3377): Remove after assert_that in streaming is fixed.
-    self.assertEqual([('a1', Timestamp(9), ['s1']),
-                      ('a2', Timestamp(9), ['s1']),
-                      ('a3', Timestamp(9), ['s1']),
-                      ('a4', Timestamp(9), ['s1']),
-                      ('b', Timestamp(9), ['s1']),
-                      ('c', Timestamp(18), ['s2'])], result)
+    assert_that(records, equal_to(
+        [TestWindowedValue(*args) for args in
+         [(('a1', Timestamp(9), ['s1']),
```

Comment:
Since you are now reifying windows, there is no need for using `RecordFn()`. They are more or less accomplishing the same things.

(This comment also applies to the previous case.

Reply:
RecordFn() is needed in that test as it is helping incorporate side inputs (AsList).
`records = (main_stream | beam.ParDo(RecordFn(), beam.pvalue.AsList(side_stream)))`
```SIDE                                 MAIN
[1, 2, 3]                             
|                                     |   
Create                             TestStream
|                                     |
Map                                WindowInto
|                                     |
WindowInto                            |          
|                  AsList             |   
_____________________________  ParDo (RecordFn)
                                      |
                                      |    (assert_that below)
         [(None, None)]            WindowInto
                |                     |
              Create                 Map
                |                     |
                __________________  CoGBK
                                      |
                                     Map
                                      |
                                     Map
```